syntax="proto3";
package brpc.policy;
option java_package="com.brpc.policy";
option java_outer_classname="BaiduRpcProto";

message RpcMeta {
    RpcRequestMeta request = 1;
    RpcResponseMeta response = 2;
    int32 compress_type = 3;
    int64 correlation_id = 4;
    int32 attachment_size = 5;
    bytes authentication_data = 7;
    int64 uid = 9;
    int32 plain_data_type = 10;
    int64 rpc_version = 101;
    bool pb_byte_base64codec_disable = 102;
    bool pb_json_map_google_pb3_format_disable = 103;
}

message DataWithType {
    uint32 type = 1;
    bytes string_data = 2;
    uint64 int_data = 3;
}

message KeyValue {
    bytes key = 1;
    DataWithType value = 2;
}

message RpcRequestMeta {
    string service_name = 1;
    string method_name = 2;
    int64 log_id = 3;
    string trace_id = 4;
    int64 span_id = 5;
    int64 parent_span_id = 6;

    int64 uri = 101;
    string svr_name = 102;
    int32 svr_gid = 103;
    string protocol_name = 104;
    string bigo_trace_id = 105;
    string bigo_span_id = 106;
    string bigo_baggage = 107;
    string bigo_flags = 108;
    repeated KeyValue key_value = 109;
}

message RpcResponseMeta {
    int32 error_code = 1;
    string error_text = 2;
    int64 next_span_id = 3;

    int32 logic_error_code = 100;
    string logic_error_text = 101;
    string bigo_next_span_id = 102;
}


